### PR TITLE
Feat(tsql): improve transpilation of temp table DDLs

### PR DESCRIPTION
**Pull request by GeorgeSittas**
@dmoore247 this covers most of the tests you wrote, but I haven't done something about table vars yet. Could continue in a future PR or feel free to take a stab yourself too.

Also considered creating a couple of new expressions instead of adding new args in `Identifier`, but the resulting code with this approach looked a bit cleaner I think. Don't feel strongly about it, so happy to pivot if needed.

### CODE REVIEW

**Comment by dmoore247** [comment]
Local and Global temp table support is huge. TSQL and Snowflake report temp table variables but I haven't encountered them much. Doing these later is good.

File: sqlglot/dialects/tsql.py [hunk 1]
```diff
@@ -518,6 +514,36 @@ def _parse_user_defined_function(
             expressions = self._parse_csv(self._parse_function_parameter)
             return self.expression(exp.UserDefinedFunction, this=this, expressions=expressions)
 
+        def _parse_id_var(
+            self,
+            any_token: bool = True,
+            tokens: t.Optional[t.Collection[TokenType]] = None,
+        ) -> t.Optional[exp.Expression]:
```
Comment (tobymao) [inline sqlglot/dialects/tsql.py:521]: is_temporary = self._match(TokenType.Hash)
is_global = is_temporary and  self._match(TokenType.Hash)

File: sqlglot/generator.py [hunk 2]
```diff
@@ -2426,6 +2427,9 @@ def expressions(
         result_sqls = []
         for i, e in enumerate(expressions):
             sql = self.sql(e, comment=False)
+            if not sql:
```
Comment (tobymao) [inline sqlglot/generator.py:2430]: why was this never hit before?

Comment (GeorgeSittas) [reply sqlglot/generator.py:2430]: Probably didn't have a use case where an expression was generating an empty string.

Comment (tobymao) [reply sqlglot/generator.py:2430]: what hits it now?

Comment (GeorgeSittas) [reply sqlglot/generator.py:2430]: The test case I tried was actually mis-formatted due to this line: https://github.com/tobymao/sqlglot/pull/1958/files#diff-362498362ae539e216cd83ebfe510977107648742f1a1bee7b2db7b5b7d27974L992: `TemporaryProperty` generates `""` for T-SQL and so even though `expressions` is simply `[""]` in this section, we still output a redundant space separator.

I mentioned this in a [comment](https://github.com/tobymao/sqlglot/pull/1958#discussion_r1274042732) above, and you can easily see that if we had more properties in the DDL we'd hit the same issue due to the `sep.join(...)` in the `expressions` method.

**Review by tobymao — APPROVED** [review]

**Pull request merged**